### PR TITLE
chore(sources): Wire `ShutdownSignal` to multiple sources with `take_until`

### PR DESCRIPTION
**Pull request by ktff**
ref. #2108 

Wires `ShutdownSignal` to  
 - `kafka` source
 - `prometheus` source
 - `statsd` source
 - `syslog` source
 - `socket` source

### CODE REVIEW

**Comment by ktff** [comment]
Also, @a-rodin it seems to me that https://github.com/fede1024/rust-rdkafka/issues/85#issuecomment-439141656 has been solved so we could possibly remove the `OwningHandle` trick?

**Comment by binarylogic** [comment]
Just verifying, this should not have any effect on performance, correct?

**Comment by ktff** [comment]
@binarylogic Yes, it shouldn't.

**Comment by binarylogic** [comment]
@a-rodin is on vacation this week, but I assume @lukesteensen or @Hoverbear can help out.

File: src/sources/util/unix.rs [hunk 1]
```diff
@@ -58,6 +63,7 @@ pub fn build_unix_source(
                 let received_from: Option<Bytes> =
                     path.map(|p| p.to_string_lossy().into_owned().into());
                 let lines_in = FramedRead::new(socket, LinesCodec::new_with_max_length(max_length))
+                    .take_until(shutdown.clone())
```
Comment (Hoverbear) [inline src/sources/util/unix.rs:66]: Seems this is an extra clone? The one above is necessary, but I don't see it used later.

Comment (ktff) [reply src/sources/util/unix.rs:66]: It is needed, since we are accepting multiple connections/sockets here.

**Pull request merged**